File: algorithms/codewars/dont-give-me-five/dont_give_me_five.py
class Solution:
    def dont_give_me_five(self, start:int, end:int) -> int:
        """
        count the number of integers that do not contain the digit 5

        Args:
            start (int): start of the range
            end (int): end of the range

        Returns:
            int: number of integers that do not contain the digit 5
        """
        return sum('5' not in str(i) for i in range(start, end+1))
    
    def dont_give_me_five_2(self, start:int, end:int) -> int:
        """
        count the number of integers that do not contain the digit 5

        Args:
            start (int): start of the range
            end (int): end of the range

        Returns:
            int: number of integers that do not contain the digit 5
        """
        count = 0
        for i in range(start, end+1):
            if '5' not in str(i):
                count += 1
        return count

File: algorithms/codewars/dont-give-me-five/dont_give_me_five.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    static int dont_give_me_five(int start, int end) {
        int count = 0;
        for (int i = start; i <= end; i++) {
            if (to_string(i).find('5') == string::npos) {
                count++;
            }
        }
        return count;
    }
};

int main(int argc, char **argv) {
    if (argc != 3) {
        cout << "Usage: ./dont_give_me_five <start> <end>" << endl;
        return 1;
    }
    int start = strtol(argv[1], NULL, 10), 
        end = strtol(argv[2], NULL, 10);
    
    if (end < start) {
        int temp = start;
        start = end;
        end = temp;
    }

    cout << Solution::dont_give_me_five(start, end) << endl;

    return 0;
}